DO_NOT_MERGE task/v2

bilalesi opened the pull request:
**DO  NOT MERGE**

bilalesi committed:
add react query

bilalesi committed:
v2 starting point

bilalesi committed:
migrate workspace shell and bit of explore

bilalesi committed:
redirect to v2 (#698)

* redirect temporarly to v2
* update setup workspace on login/redirect
* update subscription process, profile
* update stripe forms

tolokoban reviewed (dismissed):
Big an great job here, Bilal!

The code is cleaner now and the UI easier to undersand.

TanStack Query is a great asset.
Though, personally, I wouldn't have used `useQuery()` directly inside components.
I would have rather created custom hooks to abstract the data provider. Just to lower the dependency on this library that could be replaced in the future if something better emerges.

bilalesi commented on src/ui/segments/project/balance.tsx:
```diff
@@ -0,0 +1,45 @@
+  const breakpoint = useDefaultBreakpoint();
+  const { virtualLabId, projectId } = useWorkspace();
+
+  const { data, isLoading, isError, isSuccess, error } = useQuery({
```
i totally agree, I would do that too, the reason i did not, is to have v2 as quickly as possible, 
i think if we have it less then the time expected, we can ask for few days to refactor those

bilalesi committed:
add ai-assistant and fix arrows (#700)

bilalesi committed:
update counters for entities (#704)

fix brain region tree scroll issue
fix layout for ai assistant expand

bilalesi committed:
fix atlas error boundary when change brain region (#705)

bilalesi committed:
allow extensible types other than default from entitycore (#707)

bilalesi committed:
allow extensible types other than default from entitycore (use dict rather enum) (#709)

loris-olivier-obi committed:
Added the full guide skeleton with overview, tutorials and glossary (#712)

bilalesi committed:
add activities table in project home page (#715)

bilalesi committed:
add explore browse page - part-1 (#714)

* add explore browse page
* add library layout

loris-olivier-obi committed:
Help features about ai tools (#716)

* Added the content for each features

* Added AI Tools

* Clean query params when changing the main tabs

* Fixed the render of ai-tools in content.tsx

* Updated the pmpm-lock.yaml

* Added icon to code block and fixed pnpm installation

* Changed to Zod, use server side rendering when possible

* Place queryKeys into ui/use-query-keys/help.ts

* Added prefix to the help query-keys in ui

bilalesi committed:
feat/explore bookmarks (#717)

* add library listing page

bilalesi committed:
feat: mini detail view (except circuit) (#736)

* add library listing page

* add library listing page

* add mini detail view feature

bilalesi committed:
feat: cell composition  (#738)

* add cell composition to the main part of explore

g-bar committed:
New UX:  Detail view layout. (#727)

bilalesi committed:
feat: add workflow components, pages, configuration (#742)

* add workflow components, pages, configuration
* remove error boundary and use notifications (#743)

bilalesi committed:
remove unneeded packages (#745)

bilalesi committed:
add simulations data to explore (#748)

g-bar committed:
New ux: Detail view, overview page and conditional tabs. (#752)

bilalesi committed:
add memodel and synaptome build (#759)

loris-olivier-obi committed:
Notebook part 1 (#761)

* Refactored the Notebook Table

* Added left nav

* Implemented the double column with the correct table on the right

* Fix the action panel buttons of the table

tolokoban committed:
Integrate Ai Assistant and adapt resizing (#764)

bilalesi committed:
add single neuron simulation (#765)

bilalesi committed:
add new process of onboarding workspace (#768)

- add new process of onboarding workspace (#768)
(identity, provision, customization)
- fix some layouts spaces
- add responsiveness to activity table in project home

loris-olivier-obi committed:
New section reports (#771)

* Added reports to top-bar menu

* Added layouts and page for reports

* Added each section page of reports

* Ran prettier and linter

bilalesi committed:
- add synaptome simulation (#774)

- fix brain region/data left menu on update region
- update: /explore routes become /data
- add current_recordings to recording location config
- remove scope (project/public) from new simulation page

bilalesi committed:
merge develop 02-09 (#777)

pgetta committed:
Enable preview environment (via AWS) (#778) (#780)

* Add build dot env file for preview environment

* Add Github action

* Configure preview deployment for task/v2 branch for now

* Extend deployment env schema with preview deployment

bilalesi committed:
remove funds step as the funds will be added in project creation (#776)

bilalesi committed:
add transfer credits to home page (#783)

pgetta committed:
Disable CDN for preview deployment (#782)

g-bar committed:
New ux detail view pages, except circuit (#767)

bilalesi committed:
add circuit table to the new UI (#784)

bilalesi committed:
return back redirect to v2 (#785)

tolokoban committed:
Fix misalignement of the tools selector (#786)

bilalesi committed:
fix some validation behaviors (#791)

add setting last used workspace endpoint and logic

bilalesi committed:
refresh active subscriotion query after payment (#792)

bilalesi committed:
fix feedback home page (#793)

g-bar committed:
Circuit detail + small microcircuit (#796)

Thanks

bilalesi committed:
fix panel download for size and paths (#795)

fix publication time

bilalesi committed:
fix memodel configuration (#799)

add wallet click on admin role

g-bar committed:
Detail view circuit styling (#800)

bilalesi committed:
virtual lab menu item should open the settings (#797)

add mini-details-view  for simulation

bilalesi committed:
Fix/links build model (#801)

- fix links
- add warning in form validation in simulation
- fix: resolve mini detail view state management and improve form validation
- reset mini detail view when navigating between data scopes
- fix null/undefined checks using lodash isNil utility
- improve error messaging in simulation forms
- add proper React keys to prevent component state issues
- clean up login redirect logic and enhance tooltip styling

g-bar committed:
V2: Improve detail view styling (#802)

bilalesi committed:
- update workspace menu (#803)

- fix memodel and synaptome build

bilalesi committed:
fix mini-detail-view (#804)

bilalesi committed:
fix mini detail view animation state (#805)

fix logout page
fix alternate morphology download

bilalesi committed:
fix alterenate morho (#806)

fix profile in space switcher

g-bar committed:
New UX: Detail view, simulate button (#807)

* Rename

* div

* breadcrumbs

* Fetch entity

* breadcrumbs

* fetch on server

* Use slug

* navigation

* root redirect to overview

* actions

* copy action

* Add actoins

* working on bookmarks

* add bookmark categories

* add bookmark

* bookmark funtion

* Download

* lint

* lint

* bookmarks

* bookmarkCategory:

* Linting

* restore lefthook for now

* restore lefthook for now

* download entity

* Working on detail view

* Add overview page

* add morphology page

* add electrophysiology viz

* prettier

* entity one required

* Feedback

* minor

* remove unused routes

* Redirect to overview

* Visualization

* analysis tab

* emodel related artifacts

* ion channel to ion channel model

* emodel, memodel

* add synaptome

* clean up

* circuit

* fields

* add single neuron simulation

* results

* add synaptome

* add synaptome

* redirect detail root to overview

* adding name to pages

* name

* explore -> data

* remove double name

* add related publications tab

* main scree

* add circuit viz

* Add circuit analysis

* add related publications

* add mtype, etype to synaptome

* add related circuits

* add microcircuit

* lint

* fix analysis

* implement back

* fix circuit fields layout

* viz in overview for circuite

* tabbed component

* improve styles of related publications

* add tabs

* provenance

* improve analysis styles

* improve layout

* Default message

* move viz to overview

* ephys viewer

* working on fields

* fix render mtypes, etypes

* Add mtype etypes

* header layout

* improve analysis style

* implemente simulate action

* Clone MEModel

* simulate

* remove unused console.log

bilalesi committed:
redirect to latest workspace the user visited (#811)

james-isbister committed:
Minor visual changes to data browser left panel inline with mockups (#810)

* Minor visual changes to data browser left panel inline with mockups

bilalesi committed:
use virtual-lab prefix again (#812)

bilalesi committed:
fix contact email value for circuit (#817)

bilalesi committed:
mini-details view for circuit visualization (#819)

bilalesi committed:
update space switcher design (#821)

bilalesi committed:
add tooltip to min-detail view icon button (#822)

add downloading subtitle in download panel

bilalesi committed:
add new design requirements on space manager (#823)

bilalesi committed:
allow renaming virtual lab only for admins (#824)

g-bar committed:
New UX: Detail view review feedback (#820)

bilalesi committed:
update workflow process (#835)

- showing activities in the first screen
- user can open a new workflow

g-bar committed:
remove unneeded commas (#838)

g-bar committed:
Fix etype definition (#829)

bilalesi committed:
vlab name editing, names length (#839)

* vlab name too long

* return back to member list after successful invite/add operation

* order project in the space switcher by updated_at

g-bar committed:
Small microcircuit view campaign in read only mode and duplicate. Hide related publications. (#840)

g-bar committed:
Fix circuit link (#841)

bilalesi committed:
fix synaptome build (#842)

bilalesi committed:
fix synaptome simulation (#843)

bilalesi committed:
close management modal on space switcher down click (#844)

* close workspace switcher and modal when dropdown icon clicked
* fix overflow in single neuron simulation right side

bilalesi committed:
allow affliation to be editable (#845)
